src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Value/EnumConverter.cs

```diff
@@ -212,5 +214,22 @@ public override void Write(Utf8JsonWriter writer, T value, JsonSerializerOptions
                     break;
             }
         }
+
+        private string FormatEnumValue(string value)
+        {
+            if (value.IndexOf(EnumValueSeparator) == -1)
```

Comment:
This will break @pgovind's PR https://github.com/dotnet/roslyn-analyzers/pull/3610 
You can use `value.Contains(EnumValueSeparator, StringComparison.CurrentCulture)`

But you maybe didn't mean culture sensitive. So you can do `value.Contains(EnumValueSeparator)`